src/Catalyst.Core.Modules.Ledger.Tests/IntegrationTests/LedgerKvmTests.cs

```diff
@@ -0,0 +1,330 @@
+        private readonly IAccountRepository _fakeRepository;
+        private readonly IDeltaHashProvider _deltaHashProvider;
+        private readonly ILedgerSynchroniser _ledgerSynchroniser;
+        private readonly IMempool<TransactionBroadcastDao> _mempool;
+        private readonly IDeltaExecutor _deltaExecutor;
+        private readonly IStorageProvider _storageProvider;
+        private readonly ISnapshotableDb _stateDb;
+        private readonly ISnapshotableDb _codeDb;
+
+        public LedgerKvmTests()
+        {
+            _testScheduler = new TestScheduler();
+            _cryptoContext = new FfiWrapper();
+            _fakeRepository = Substitute.For<IAccountRepository>();
+            _hashProvider = new HashProvider(HashingAlgorithm.GetAlgorithmMetadata("blake2b-256"));
+            _genesisHash = _hashProvider.ComputeUtf8MultiHash("genesis");
+
+            _logger = Substitute.For<ILogger>();
+            _mempool = Substitute.For<IMempool<TransactionBroadcastDao>>();
+            _deltaHashProvider = Substitute.For<IDeltaHashProvider>();
+            _ledgerSynchroniser = Substitute.For<ILedgerSynchroniser>();
+
+            _ledgerSynchroniser.DeltaCache.GenesisHash.Returns(_genesisHash);
+
+            IDb stateDbDevice = new MemDb();
+            IDb codeDbDevice = new MemDb();
+
+            _stateDb = new StateDb(stateDbDevice);
+            _codeDb = new StateDb(codeDbDevice);
+
+            _stateProvider = new StateProvider(_stateDb, _codeDb, LimboLogs.Instance);
+            _storageProvider = new StorageProvider(_stateDb, _stateProvider, LimboLogs.Instance);
+
+            IStateUpdateHashProvider stateUpdateHashProvider = new StateUpdateHashProvider();
+            _specProvider = new CatalystSpecProvider();
+
+            IKvm kvm = new KatVirtualMachine(_stateProvider, _storageProvider, stateUpdateHashProvider, _specProvider, LimboLogs.Instance);
+            _deltaExecutor = new DeltaExecutor(_specProvider, _stateProvider, _storageProvider, kvm, new FfiWrapper(), _logger);
+        }
+        
+        private void RunDeltas(Delta delta)
+        {
+            var hash1 = _hashProvider.ComputeUtf8MultiHash("update");
+            var updates = new[] {hash1};
+            _ledgerSynchroniser.CacheDeltasBetween(default, default, default)
+               .ReturnsForAnyArgs(new Cid[] {hash1, _genesisHash});
+            _ledgerSynchroniser.DeltaCache.TryGetOrAddConfirmedDelta(Arg.Any<Cid>(), out Arg.Any<Delta>())
+               .Returns(c =>
+                {
+                    c[1] = delta;
+                    return true;
+                }, c => false);
+
+            _deltaHashProvider.DeltaHashUpdates.Returns(updates.Select(h => (Cid) h).ToObservable(_testScheduler));
+
+            Ledger ledger = new Ledger(_deltaExecutor, _stateProvider, _storageProvider, _stateDb, _codeDb, _fakeRepository, _deltaHashProvider, _ledgerSynchroniser, _mempool, _logger);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'ledger'.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4350015)